### PR TITLE
[Terrain] Second batch of Paint Brush unit tests - validates Paint Brush Settings

### PR DESCRIPTION
**Pull request by mbalfour-amzn**
## What does this PR do?

Adds another round of unit tests for the PaintBrush code. This verifies that the following Paint Brush Settings correctly affect painting and smoothing:
* Brush opacity
* Brush size
* Hardness %
* Flow %
* Distance %
* Blend modes
* Smoothing radius
* Smoothing modes

## How was this PR tested?

Ran the unit tests and verified they passed.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintSettingsTests.cpp [hunk 1]
```diff
@@ -0,0 +1,839 @@
...
+        {
+            m_settings.SetSize(brushRadiusSize * 2.0f);
+
+            ValidationFn validateFn = [=](const AZ::Aabb& dirtyArea,
+                                  AzToolsFramework::PaintBrushNotifications::ValueLookupFn& valueLookupFn)
+            {
+                // The dirtyArea AABB should change size based on the current brush radius size that we're using.
+                EXPECT_THAT(dirtyArea, IsClose(AZ::Aabb::CreateCenterRadius(TestBrushCenter2d, brushRadiusSize)));
+
+                // Create a 3x3 square grid of points. Because our brush is a circle, we expect only the points in along a + to be
```
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintSettingsTests.cpp:215]: The wording of `in along a +` confused me a bit, maybe just `along a +` ?

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintSettingsTests.cpp (hunk 1 above)
Comment (mbalfour-amzn) [reply Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintSettingsTests.cpp:215]: Whoops, typo, thx!

File: Code/Framework/AzToolsFramework/AzToolsFramework/PaintBrush/PaintBrush.cpp [hunk 2]
```diff
@@ -274,7 +278,7 @@ namespace AzToolsFramework
         // as the starting point.
         if (m_isFirstPointInBrushStrokeMovement)
         {
-            brushStampCenters.emplace_back(currentAdjustedCenter);
+            brushStampCenters.push_back(currentAdjustedCenter);
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/PaintBrush/PaintBrush.cpp:281]: Thanks for updating this 🙂

Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/PaintBrush/PaintBrush.cpp:281]: Bonus points for updating line 302 as well 😉

**Review by hultonha — APPROVED** [review]
Really awesome to see this incredibly thorough test coverage! 😄 Excellent work 😎

**Pull request merged**